spectrocloud/resource_cluster_libvirt.go

```diff
@@ -685,6 +708,20 @@ func flattenMachinePoolConfigsLibvirt(machinePools []*models.V1LibvirtMachinePoo
 			pj["image_storage_pool"] = p.SourceStoragePool
 			pj["target_storage_pool"] = p.TargetStoragePool
 			pj["data_storage_pool"] = p.DataStoragePool
+			dConfig := make([]interface{}, 0)
+			if p.GpuDevices != nil {
+				gpuDevices := p.GpuDevices
+				for _, d := range gpuDevices {
+					if !(d.Model == "" || d.Vendor == "") {
+						dElem := make(map[string]interface{})
+						dElem["device_model"] = d.Model
+						dElem["vendor"] = d.Vendor
+						dElem["addresses"] = d.Addresses
+						dConfig = append(dConfig, dElem)
+					}
+				}
+			}
+			pj["gpu_device"] = dConfig
```

Comment:
Should not assign if config is nil

Reply:
@SivaanandM let's move it inside if statement so it's not set if nil.

Reply:
this updated